### ISSUE TITLE
CA2018: Handle Arrays passed in using Array.CreateInstance

### ISSUE THREAD

**Issue by mahdiva** [dotnet/roslyn-analyzers#5371]
### Analyzer

**Diagnostic ID**: [CA2018](https://github.com/dotnet/roslyn-analyzers/pull/5242): `'Buffer.BlockCopy' expects the number of bytes to be copied for the 'count' argument`

### Describe the bug

CA2018 currently does not check to throw a diagnostic for arrays created using `Array.CreateInstance`. The Roslyn API doesn't seem to provide the array element type in order to check if the array is a not of type `byte` or `sByte` to throw a diagnostic.

[Link to the PR conversation](https://github.com/dotnet/roslyn-analyzers/pull/5242#discussion_r686126730)

There is a unit test available for this, however it's currently being skipped

### Steps To Reproduce

```csharp
using System;

class Program
{
    static void Main()
    {
        Array src = Array.CreateInstance(typeof(int), 4);
        Array dst = Array.CreateInstance(typeof(int), 4);

        Buffer.BlockCopy(src, 0, dst, 0, src.Length);
    }
}
```

### Expected behavior

There should be a diagnostic for `src.Length` since the array element is of type `int`

### Actual behavior

No CA2018 diagnostic is reported in this case
